File: android/app/src/main/kotlin/com/sps/patrol/gps/SuperGpsMethodHandler.kt
package com.sps.patrol.gps

import android.content.Context
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class SuperGpsMethodHandler(private val context: Context) : MethodChannel.MethodCallHandler {
    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "getCurrentPosition" -> {
                val enableBarometer = call.argument<Boolean>("enableBarometer") ?: false
                SuperGpsLocationEngine.getCurrentPosition(
                    context.applicationContext,
                    enableBarometer,
                    result,
                )
            }
            "isBarometerSupported" -> {
                val supported = SuperGpsLocationEngine.isBarometerHardwareSupported(
                    context.applicationContext,
                )
                result.success(supported)
            }
            else -> result.notImplemented()
        }
    }
}

File: ios/Runner/SuperGpsMethodHandler.swift
import Flutter

final class SuperGpsMethodHandler: NSObject {
    func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "getCurrentPosition":
            let args = call.arguments as? [String: Any]
            let enableBarometer = args?["enableBarometer"] as? Bool ?? false
            SuperGpsLocationEngine.shared.getCurrentPosition(
                enableBarometer: enableBarometer,
                result: result
            )
        case "isBarometerSupported":
            result(SuperGpsLocationEngine.shared.isBarometerHardwareSupported())
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}
